specs/governance/gov-token.md

```diff
@@ -24,72 +24,74 @@
 
 `GovernanceToken` is an [ERC20](https://eips.ethereum.org/EIPS/eip-20) token contract that inherits from `ERC20Burnable`,
 `ERC20Votes`, and `Ownable`. It allows token holders to delegate their voting power to other addresses, enabling a representative
-voting system.
+voting system. The contract integrates with the `Alligator` contract through a hook-based approach to enable advanced delegation
+functionality, including partial delegation and subdelegations.
 
-The specific implementation details of the voting and delegation logic are inherited from the `ERC20Votes` contract. The
-`GovernanceToken` contract focuses on integrating this functionality with `ERC20Burnable` and adding the minting
-capability. EIP-712 permit functionality is also inherited from `ERC20Permit`, which is imported by `ERC20Votes`.
+### Hook-based Integration with Alligator
 
-### Checkpoints
+The `GovernanceToken` contract integrates with the `Alligator` contract through a hook-based approach to enable advanced
+delegation features. The `_afterTokenTransfer` function in the `GovernanceToken` is modified to call the `afterTokenTransfer`
+function in the `Alligator` contract, allowing the `Alligator` contract to consume the hooks and update its delegation and checkpoint
+mappings accordingly.
 
-`ERC20Votes` uses checkpoints to track voting power history. A checkpoint is a "snapshot" of the voting power of an
-address at a specific block number.
+If the call to the Alligator's `afterTokenTransfer` function fails, the token transfer must still be completed, and an
+`AlligatorCallFailed` event will be emitted to indicate the failed call.
 
 ```solidity
-struct Checkpoint {
-    uint32 fromBlock;
-    uint224 votes;
-}
+emit AlligatorCallFailed(from, to, amount);
```

Comment:
I named the params `transferFrom` and `transferTo` to make it clear it's the accounts involved in the token transfer, instead of being interpreted as the from and to address of the failed call
```suggestion
event AlligatorCallFailed(address indexed transferFrom, address indexed transferTo, uint256 amount);
```

Reply:
good idea with this!